src/communication/communicator.hpp

```diff
@@ -128,7 +128,7 @@ class communicator {
     /// The range of event queues that belong to cells in group i.
     std::pair<cell_size_type, cell_size_type> group_queue_range(cell_size_type i) {
         EXPECTS(i<num_local_groups_);
-        return {domain_id_partition_[i], domain_id_partition_[i+1]};
+        return index_part_[i];
```

Comment:
Or even just expose (directly or through a method) the partition view itself. Just a matter of taste at this point though.